helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -552,7 +551,6 @@ protected synchronized Set<HelixConstants.ChangeType> doRefresh(HelixDataAccesso
 
     updateIdealRuleMap(getClusterConfig());
     updateDisabledInstances(getInstanceConfigMap().values(),
```

Comment:
this is the cache which required update?

Reply:
Yes, in node swap cache changes we introduced `_assignableDisabledInstanceSet`. We were not clearing it on cache refresh. This would lead WagedCapacityProvider to not charge pending messages against the instances capacity for any instance that was offline at some point.

After revising, this cache is not necessary as we can use `_disabledInstanceSet` to determine enabled instances and enabled assignable instances since it will be a superset of `_assignableDisabledInstanceSet`.